### PR TITLE
Added two metrics about memory usage in cgroup to asynchronous metrics

### PR DESCRIPTION
**Pull request by sichenzhao**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
- Add total memory and used memory metrics with respect to cgroup in AsyncMetrics (https://github.com/ClickHouse/ClickHouse/issues/37983)

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)
-- Updated document

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Common/AsynchronousMetrics.cpp [hunk 1]
```diff
@@ -879,6 +882,31 @@ void AsynchronousMetrics::update(TimePoint update_time)
...
+            cgroupmem_limit_in_bytes->rewind();
+            cgroupmem_usage_in_bytes->rewind();
+
+            uint64_t cgroup_mem_limit_in_bytes = 0;
+            uint64_t cgroup_mem_usage_in_bytes = 0;
+
+            readText(cgroup_mem_limit_in_bytes, *cgroupmem_limit_in_bytes);
+            readText(cgroup_mem_usage_in_bytes, *cgroupmem_usage_in_bytes);
+
+            if (!cgroup_mem_limit_in_bytes || !cgroup_mem_usage_in_bytes) {
```
Comment (qoega) [inline src/Common/AsynchronousMetrics.cpp:897]:
```suggestion
            if (!cgroup_mem_limit_in_bytes || !cgroup_mem_usage_in_bytes)
            {
```

**Review by nikitamikhaylov — APPROVED** [review]

**Comment by nikitamikhaylov** [comment]
A brief description for these two new metrics is enough.

**Comment by nikitamikhaylov** [comment]
Integration tests are broken in master also and not related to this PR.

**Comment by alexey-milovidov** [comment]
@sichenzhao why don't I have these files? https://pastila.nl/?00261dc3/75cbe14570eb6ac77ee292a05eeb1557

**Comment by sichenzhao** [comment]
> @sichenzhao why don't I have these files? https://pastila.nl/?00261dc3/75cbe14570eb6ac77ee292a05eeb1557

docker desktop uses cgroup-v2 by default, that's probably why. This change seems to be working for cgroup_v1 only. let me add follow up PR for cgroup_v2 support later

**Pull request merged**